client/lib/wp/localization/index.js

```diff
@@ -43,9 +43,17 @@ export function addLocaleQueryParam( params ) {
 		return params;
 	}
 
+	let localeQueryParam = {};
 	const query = qs.parse( params.query );
+
+	if ( 'wpcom/v2' === params.apiNamespace ) {
```

Comment:
Hmm. `wpcom/v2` isn't the only namespace that will use the core API, for example, `wp/v2`:

https://public-api.wordpress.com/wp/v2/sites/example.wordpress.com/types?_locale=fr-FR

(Aside: Unsure why that wouldn't be translated ^)

By searching how `apiNamespace` is handled, it seems it might be exclusively used for core API:

https://github.com/Automattic/wpcom.js/search?utf8=%E2%9C%93&q=apinamespace&type=

In which case it might be fair enough to simply test truthiness of `params.apiNamespace` to decide to prefix the underscore.

cc @nylen @timmyc